packages/driver-kube-pod/src/driver/client/metadata.ts

```diff
@@ -129,8 +130,30 @@ export const profileSelector = ({ profileId }: { profileId: string }) => ({
 export const isDockerHostDeployment = (s: k8s.KubernetesObject) => s.kind === 'Deployment'
     && s.metadata?.labels?.[LABELS.COMPONENT] === DOCKER_HOST_VALUE
 
+const MAX_NAME_LENGTH = 63
+
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+const sanitizeName = (s: string) => truncateWithHash(
```

Comment:
If you want, you can check your generated name against this RegEx (taken from k8s log):

```
[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```